src/gluonts/dataset/split/splitter.py

```diff
@@ -71,47 +71,70 @@ class Config:
     feat_dynamic_real: List[pd.Series] = []
 
     @classmethod
-    def from_time_series_item(
-        cls, item: TimeSeriesItem, freq: Optional[str] = None
+    def from_data_entry(
+        cls, item: DataEntry, freq: Optional[str] = None
     ) -> "TimeSeriesSlice":
         if freq is None:
-            freq = item.start.freq
+            freq = item["start"].freq
 
         index = pd.date_range(
-            start=item.start, freq=freq, periods=len(item.target)
+            start=item["start"], freq=freq, periods=len(item["target"])
         )
 
+        item_dynamic_cat = (
+            item["feat_dynamic_cat"] if "feat_dynamic_cat" in item else []
+        )
         feat_dynamic_cat = [
-            pd.Series(cat, index=index) for cat in item.feat_dynamic_cat
+            pd.Series(cat, index=index) for cat in item_dynamic_cat
         ]
 
+        item_dynamic_real = (
+            item["feat_dynamic_real"] if "feat_dynamic_real" in item else []
+        )
         feat_dynamic_real = [
-            pd.Series(real, index=index) for real in item.feat_dynamic_real
+            pd.Series(real, index=index) for real in item_dynamic_real
         ]
 
+        feat_static_cat = (
+            list(item["feat_static_cat"]) if "feat_static_cat" in item else []
+        )
```

Comment:
I think this is a bit more concise:

```suggestion
        feat_static_cat = list(item.get("feat_static_cat", []))
```